tycho-ethereum/src/rpc/mod.rs

```diff
@@ -0,0 +1,856 @@
+use std::{collections::HashMap, default::Default, time::Duration};
+
+use alloy::{
+    primitives::{Address, B256, U256},
+    rpc::{
+        client::{ClientBuilder, ReqwestClient},
+        types::{
+            debug::StorageRangeResult,
+            trace::parity::{TraceResults, TraceType},
+            Block, BlockId, BlockNumberOrTag, TransactionRequest,
+        },
+    },
+    transports::http::reqwest,
+};
+use tracing::{debug, info, trace};
+use tycho_common::Bytes;
+
+use crate::{errors::extract_error_chain, RPCError, RequestError};
+
+// TODO: Optimize these configurable for preventing rate limiting.
+// TODO: Handle rate limiting / individual connection failures & retries
+#[derive(Clone, Debug)]
+pub struct BatchingConfig {
+    max_batch_size: usize,
+    max_storage_slot_batch_size: usize,
+}
+
+impl Default for BatchingConfig {
+    fn default() -> Self {
+        Self { max_batch_size: 50, max_storage_slot_batch_size: 10000 }
+    }
+}
+
+// TODO: Consider adding rate limiting and retry logic here
```

Comment:
I think retry logic is very important, fine if it goes on a separate PR but it's a huge improvement